### PR TITLE
Add specifiedBy default directive

### PR DESCRIPTION
**Pull request by swalkinshaw**
SDL schemas that use a `specifiedBy` directive but _don't_ include the directive definition will fail when using `GraphQL::Schema::BuildFromDefinition`. Because `specifiedBy` is a built-in directive (in the draft spec at least), this should be supported.

This PR just defines the directive and includes it as a default directive.

### CODE REVIEW

**Comment by swalkinshaw** [comment]
Ran my tests before adding this to default directives so I'll update these

**Comment by rmosolgo** [comment]
:+1: Thanks for this addition, I didn't realize it had been added to the spec. 

I'm game to add it, but I think we should harmonize it with existing `specifiedByURL` support. I think these should be updated to read/write using directives:

https://github.com/rmosolgo/graphql-ruby/blob/9838bbee177448b91acea197eb48d157a4152bc6/lib/graphql/schema/scalar.rb#L20-L30

https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/schema/loader.rb#L145

But this one should work fine as-is: 

https://github.com/rmosolgo/graphql-ruby/blob/9838bbee177448b91acea197eb48d157a4152bc6/lib/graphql/schema/build_from_definition.rb#L306

**Comment by swalkinshaw** [comment]
Should `specified_by_url` even exist on the the scalar base type? Wouldn't the equivalent now be:

```ruby
class MyCustomScalar < GrpahQL::Schema::Scalar
  directive GraphQL::Schema::Directive::SpecifiedBy, url: "http://example.com"
end
```

Guess it depends if you want to keep it for compatibility

**Comment by rmosolgo** [comment]
Yes, agreed it could be migrated to `directive ...`. But I don't know about downstream usage, so could we keep `specified_by_url` as a shorthand for that, for compatibility?

**Comment by swalkinshaw** [comment]
👍  gave it a shot

File: lib/graphql.rb [hunk 1]
```diff
@@ -101,8 +101,8 @@ module EmptyObjects
 require "graphql/pagination"
 require "graphql/schema"
 require "graphql/query"
-require "graphql/types"
 require "graphql/dataloader"
+require "graphql/types"
```
Comment (swalkinshaw) [inline lib/graphql.rb:105]: I ran into an issue with `GraphQL::Dataloader::NullDataloader` not being defined at this point due to referencing `Directive` within `Scalar`

File: lib/graphql/schema/scalar.rb [hunk 2]
```diff
@@ -19,9 +19,9 @@ def kind
 
         def specified_by_url(new_url = nil)
           if new_url
-            @specified_by_url = new_url
-          elsif defined?(@specified_by_url)
-            @specified_by_url
+            directive(GraphQL::Schema::Directive::SpecifiedBy, url: new_url)
+          elsif (directive = directives.find { |dir| dir.graphql_name == "specifiedBy" })
+            directive.arguments[:url] # rubocop:disable Development/ContextIsPassedCop
```
Comment (swalkinshaw) [inline lib/graphql/schema/scalar.rb:24]: `Directive.arguments` is just an attr reader from what I can see but this cop doesn't know that

**Comment by rmosolgo** [comment]
Thanks for this improvement!

**Pull request merged**